includes/admin/class-wc-rest-stripe-account-keys-controller.php

```diff
@@ -335,6 +357,94 @@ public function test_account_keys( WP_REST_Request $request ) {
 		return new WP_REST_Response( [], 200 );
 	}
 
+	/**
+	 * Configure webhooks for the Stripe Account.
+	 *
+	 * This will create a webhook endpoint in the Stripe account with the events we need to listen to at the correct URL.
+	 * The webhook secret will be stored in the settings.
+	 *
+	 * @param WP_REST_Request $request Data about the request.
+	 */
+	public function configure_webhooks( WP_REST_Request $request ) {
+		$settings     = get_option( self::STRIPE_GATEWAY_SETTINGS_OPTION_NAME, [] );
+		$live_mode    = wc_clean( wp_unslash( $request->get_param( 'live_mode' ) ) );
+		$secret       = wc_clean( wp_unslash( $request->get_param( 'secret' ) ) );
+		$saved_secret = $settings[ $live_mode ? 'secret_key' : 'test_secret_key' ];
+
+		// Check if the user is configuring the opposite mode. ie if the store is in live mode and is configuring webhooks for test mode.
+		$is_testmode_enabled     = 'yes' === $settings['testmode'];
+		$configure_opposite_mode = isset( $settings['testmode'] ) && ( $live_mode && $is_testmode_enabled ) || ( ! $live_mode && ! $is_testmode_enabled );
```

Comment:
Would the `isset( $settings['testmode'] )` check be needed here if we're doing `$is_testmode_enabled = 'yes' === $settings['testmode'];` in the line above?

Also, wouldn't it be an error if `$settings['testmode']` isn't set before reaching this line?

Reply:
Yeah good point. I've rejigged this [41267c3](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3209/commits/41267c37756cb32aa5672c3e528eb29c5ac06ce0). I now check if test mode is set in the `$is_test_mode_enabled` definition and then removed it from the next line.